components/server/src/errors/index.ts

```diff
@@ -5,62 +5,38 @@
  */
 
 import { User, Token } from "@gitpod/gitpod-protocol";
+import {
+    RepositoryNotFoundError,
+    UnauthorizedRepositoryAccessError,
+} from "@gitpod/gitpod-protocol/lib/messaging/error";
 
-export interface NotFoundError extends Error {
-    readonly data: NotFoundError.Data;
-}
 export namespace NotFoundError {
-    export interface Data {
-        readonly host: string;
-        readonly owner: string;
-        readonly repoName: string;
-        readonly userScopes?: string[];
-        readonly lastUpdate?: string;
-    }
     export async function create(token: Token | undefined, user: User, host: string, owner: string, repoName: string) {
-        const lastUpdate = token && token.updateDate;
+        const lastUpdate = (token && token.updateDate) || "";
         const userScopes = token ? [...token.scopes] : [];
 
         const userIsOwner = owner == user.name; // TODO: shouldn't this be a comparison with `identity.authName`?
-        const data = <NotFoundError.Data>{ host, owner, repoName, userIsOwner, userScopes, lastUpdate };
-        const error = Object.assign(new Error("NotFoundError"), { data });
-        return error;
+        return new RepositoryNotFoundError({
+            host,
+            owner,
+            userIsOwner,
+            userScopes,
+            lastUpdate,
+        });
     }
-    export function is(error: any): error is NotFoundError {
-        return (
-            !!error &&
-            !!error.data &&
-            !!error.data.host &&
-            !!error.data.owner &&
-            !!error.data.repoName &&
-            error.message === "NotFoundError"
-        );
+    export function is(error: any): error is RepositoryNotFoundError {
+        return error instanceof RepositoryNotFoundError;
     }
 }
 
-export interface UnauthorizedError extends Error {
-    readonly data: UnauthorizedError.Data;
-}
 export namespace UnauthorizedError {
-    export interface Data {
-        readonly host: string;
-        readonly scopes: string[];
-        readonly messageHint: string;
-    }
-    const message = "UnauthorizedError";
     export function create(host: string, scopes: string[], messageHint?: string) {
-        const data = <UnauthorizedError.Data>{ host, scopes, messageHint: messageHint || "unauthorized" };
-        const error = Object.assign(new Error(message), { data });
-        return error;
+        return new UnauthorizedRepositoryAccessError({
```

Comment:
Thanks for providing a smooth transitioning path! 🙇🏻 
This feels great! Not touching the call sites but still converting 👍🏻